src/main/java/study/calculator/Numbers.java

```diff
@@ -0,0 +1,52 @@
+package study.calculator;
+
+import study.calculator.pattern.CustomPattern;
+import study.calculator.pattern.DefaultPattern;
+import study.calculator.pattern.PatternSeparator;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+import static study.calculator.Validator.PREFIX_STRATEGY_CONDITION;
+import static study.calculator.Validator.STRING_WHITESPACE;
+
+/**
+ * 숫자를 관리 할 일급 컬렉션
+ */
+public class Numbers {
+
+    private final List<Number> numbers;
+
+    public Numbers() {
+        this(STRING_WHITESPACE);
+    }
+
+    private Numbers(String text) {
+        this(text, new DefaultPattern());
+    }
+
+    protected Numbers(String text, PatternSeparator patternSeparator) {
+        numbers = getExtractNumbers(text, patternSeparator);
+    }
+
+    public static Numbers of(String text) {
+        if(text.startsWith(PREFIX_STRATEGY_CONDITION)) {
+            return new Numbers(text, new CustomPattern());
+        }
+        return new Numbers(text);
+    }
+
+    protected List<Number> getExtractNumbers(String text, PatternSeparator patternSeparator) {
+        String[] stringsSeparatedBySeparator = patternSeparator.matches(text);
+
+        return Arrays.stream(stringsSeparatedBySeparator)
+                .filter(Validator::isNumeric)
+                .map(Number::of)
+                .collect(toList());
+    }
+
+    public List<Number> getNumbers() {
+        return numbers;
+    }
```

Comment:
일급컬렉션을 도입했으니 최대한 getter없이 일급컬렉션만을 사용하도록 만드는게 어떨까요? 🤔 
getNumbers를 사용하는 곳에서 숫자들의 합을 구하는 곳에 사용하고 있으니 아래와같은 메서드를 제공하는 것이 캡슐화에도 더 도움이 될것 같아요. 😁 

(테스트코드에서도 사용하고 있네요. containsExactly()를 사용하고 싶으시다면 일급컬렉션에서 Iterable 인터페이스를 구현해보세요!)

```suggestion
        public int sumNumbers() {
        return numbers.stream()
                .map(Number::getNumber)
                .reduce(Integer::sum)
                .orElse(0);
    }
```

Reply:
앞으로 getter를 최대한 사용하지 않고 의미 있는 메서드로 수정해보겠습니다.